### PR TITLE
Upstream up-to 2020-09-22 - non-controversial parts

### PR DESCRIPTION
**Pull request by amilcarlucas**
I hope these are not very controversial :)

### CODE REVIEW

**Comment by hamishwillee** [comment]
Looks pretty good to me. I particularly like the changes to capitalisation of North, South etc. Was never really happy with that.

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -916,19 +916,19 @@
         <param index="2">Empty</param>
         <param index="3">Empty</param>
         <param index="4">Empty</param>
-        <param index="5">Latitude</param>
-        <param index="6">Longitude</param>
-        <param index="7">Altitude</param>
+        <param index="5" label="Latitude" units="degE7">Latitude of ROI location</param>
+        <param index="6" label="Longitude" units="degE7">Longitude of ROI location</param>
```
Comment (peterbarker) [inline message_definitions/v1.0/common.xml:920]: Aren't the units just "deg" when this is transported by `COMMAND_LONG`?

**Comment by peterbarker** [comment]
My take on these:

```
common.xml: Add missing MAV_CMD units - does more than that, changing descriptions which ArduPilot may or may not conform with; REMOVE
remove wip status from GENERATOR_STATUS - OK
Move winch and gripper commands from ardupilotmega to common - OK
Include minimal in common.xml - does more than that, changing lots of stuff; REMOVE
Update to indexed name support - OK
move MAV_CMD_FIXED_MAG_CAL_YAW ardupilotmega -> common - OK
scripts: re-enable format_xml call - OK
common.xml: spelling and small misc corrections  - OK
common.xml: deprecate some messages in favor of SET_MESSAGE_INTERVAL - deprecates DATA_STREAM which is still the primary way GCSs interact with ArduPilot.  This could probably still go in as there is a better way, but it deserves a separate PR. - REMOVE
common.xml: improve wording - query about the correctness on units - REMOVE
```

By "REMOVE" I mean "remove from this PR; they might still be good changes (the "extra stuff" in the mav-cmd-units patch), but they need special checking and are thus non-controversial :-)

**Comment by amilcarlucas** [comment]
Rebased on top of master with removal of some patches.

**Review by tridge — APPROVED** [review]

**Review by peterbarker — APPROVED** [review]

**Pull request merged**